### PR TITLE
Assign PBR textures mel script

### PR DESCRIPTION
**Pull request by shizorr**

### CODE REVIEW

File: maya/mel/assign-pbr-textures.mel [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{
+    $sl = {""};
```
Comment (Ziriax) [inline maya/mel/assign-pbr-textures.mel]: Why is this line needed?

File: maya/mel/assign-pbr-textures.mel [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    $sl = `ls -sl`;
+    
+    if($sl[0]=="")
```
Comment (Ziriax) [inline maya/mel/assign-pbr-textures.mel]: `size($sl)==0` seems a better way to check for an empty array

File: maya/mel/assign-pbr-textures.mel [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    {
+        print "No selection to apply the material to...\n";
+    }
+    else
+    {
+        loadPlugin "C:/Program Files/Autodesk/Maya2017/bin/plug-ins/glslShader.mll";
```
Comment (Ziriax) [inline maya/mel/assign-pbr-textures.mel]: Is the absolute path needed? I think you can just pass `glslShader.mll`. Could you try this?

File: maya/mel/assign-pbr-textures.mel [hunk 4]
```diff
@@ -0,0 +1,43 @@
+{
```
Comment (Ziriax) [inline maya/mel/assign-pbr-textures.mel]: Might be a good idea to place this code in a MEL procedure? E.g. `assign_glTF_PBR_material_and_textures`? Passing the node id as a parameter, using the selection if it is the empty string?

File: maya/mel/assign-pbr-textures.mel [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    {
+        print "No selection to apply the material to...\n";
+    }
+    else
+    {
+        loadPlugin "C:/Program Files/Autodesk/Maya2017/bin/plug-ins/glslShader.mll";
+        source "createAndAssignShader.mel";
+    
+        $sh = `shadingNode -asShader GLSLShader`;
+        setAttr -type "string" ($sh+".shader") "D:\\Development\\PBR\\shaders\\glTF_PBR.ogsfx";
```
Comment (Ziriax) [inline maya/mel/assign-pbr-textures.mel]: The path to the shader should not be hardcoded, best to pass this as a required argument to the procedure.

File: maya/mel/assign-pbr-textures.mel [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        assignCreatedShader "GLSLShader" "" $sh $sl[0];
+        
+        $numAttrs = 9;
```
Comment (Ziriax) [inline maya/mel/assign-pbr-textures.mel]: `$numAttrs` is `size($attrNames)` no?

File: maya/mel/assign-pbr-textures.mel [hunk 7]
```diff
@@ -0,0 +1,43 @@
+{
+    $sl = {""};
+    $sl = `ls -sl`;
```
Comment (Ziriax) [inline maya/mel/assign-pbr-textures.mel]: In MEL, the guideline is to mention the type explicitly, e.g. `string $sel[] = `...

File: maya/mel/assign-pbr-textures.mel [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+            "u_EmissiveTexture",
+            "u_DiffuseEnvTexture",
+            "u_SpecularEnvTexture",
+            "u_brdfTexture"
+        };
+        
+        for ($i=0;$i<$numAttrs;++$i)
+        {
+            $shAttrName = ($sh+"."+$attrNames[$i]);
+            $filePath = {""};
```
Comment (Ziriax) [inline maya/mel/assign-pbr-textures.mel]: use `string $filePaths[]`

File: maya/mel/assign-pbr-textures.mel [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+            $filePath = `fileDialog2 -cap ("Open '"+$shAttrName+"'") -fm 1`;
```
Comment (Ziriax) [inline maya/mel/assign-pbr-textures.mel]: Asking for every filename is one approach. But artists will still complain about  the many dialog boxes. Another common approach in studios might be naming conventions. I would add a parameter to the procedure with these naming conventions (e.g. an array of regular expressions for each of the texture attributes). If this array is passed, then let the user select many textures files at once, and then test each file again the reg-ex, putting it in the right file node, if that reg-ex matches. Otherwise do not create a file node at all. What do you think?

File: maya/mel/assign-pbr-textures.mel [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+            if($filePath[0]!="")
```
Comment (Ziriax) [inline maya/mel/assign-pbr-textures.mel]: Use `size`

File: maya/mel/assign-pbr-textures.mel [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+            {
+                $fileNode = `createRenderNodeCB -as2DTexture "" file ("defaultNavigation -force true -connectToExisting -source %node -destination "+$shAttrName+";")`;
```
Comment (Ziriax) [inline maya/mel/assign-pbr-textures.mel]: specify type

File: maya/mel/assign_glTF_PBR_material_and_textures.mel [hunk 12]
```diff
@@ -0,0 +1,80 @@
+global proc assign_glTF_PBR_material_and_textures(
```
Comment (Ziriax) [inline maya/mel/assign_glTF_PBR_material_and_textures.mel]: I would pass the regular expressions of the texture file-names as an argument, and make a `global $default_glTF_texture_patterns` or something. We can't hard-code these texture names, this is studio and tool dependent.

File: maya/mel/assign_glTF_PBR_material_and_textures.mel [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+        "u_RoughnessTexture",
+        "u_OcclusionTexture",
+        "u_NormalTexture",
+        "u_EmissiveTexture",
+        "u_DiffuseEnvTexture",
+        "u_SpecularEnvTexture",
+        "u_brdfTexture"
+    };
+
+    string $attrExprs[] = {
```
Comment (Ziriax) [inline maya/mel/assign_glTF_PBR_material_and_textures.mel]: See first comment, this should become a parameter

**Comment by shizorr** [comment]
How do you feel about the environment maps and brdf lut map being passed as parameters as well? We would need a new script that requests the user to point to these files and save them as globals or something.
